app/models/concerns/facets/managed_host_extensions.rb

```diff
@@ -0,0 +1,50 @@
+require 'facets'
+
+module Facets
+  module ManagedHostExtensions
+    extend ActiveSupport::Concern
+
+    included do
+      Facets::ManagedHostExtensions.refresh_facet_relations(self)
+      after_save :clear_association_cache #should be removed after moving to rails 4. Fixes an issue with save! that breaks :inverse_of.
+
+      def attributes
+        hash = super
+
+        # include all facet attributes by default
+        host_facets_with_definitions.each do |facet, facet_definition|
+          hash["#{facet_definition.name}_attributes"] = facet.attributes.reject { |key, _| %w(created_at updated_at).include? key }
```

Comment:
This method doesn't seem like it exists for me, added:

```
        define_method("#{facet_config.name}_attributes") do
          self.send(facet_config.name).attributes
        end
```

in the included register_facet_relation method and that seemed to have fixed it